io.github.tralph3.Steam_Metadata_Editor.metainfo.xml

```diff
@@ -0,0 +1,368 @@
+</em>appinfo.vdf<em>
+</em> to to get the checksum.
+</p>
+<p>
+  <em>
+  </em>NOTE:<em>
+</em> The encoding of the binary data is in UTF-8 for the most part. However, some strings can&apos;t be decoded with it. For those cases, I use ISO8859-1 as a fallback, which will decode everything no matter what. This also means that I must use the same encoding when I encode the data back. To signify that I used the ISO8859-1 encoding, I append a <code>0x06</code> byte to such strings, which then gets removed when producing checksums or writing data.
+</p>
+<p>
+  Then what do we do exactly? We format all the modified data this way (the function <code>format_data</code> takes care of this in my program), and then pass the result through SHA-1, and we get the checksum. <em>However</em>, there is one final thing that made me scratch my head for months that I couldn&apos;t figure out alone. Some apps have backslashes in their data. These backslashes are usually used for directory paths and stuff like that, and as you may know, backslashes are escape characters.
+</p>
+<p>
+  So... what does this mean to us? It means that I spent months trying to figure out why I could produce valid checksums for some games, while others failed all the time. When generating the checksum, you need to add a second backslash to every single one that appears. That&apos;s why in <code>format_data</code> I use <code>replace</code> to change this: <code>\\</code> to this <code>\\\\</code>. As the backslash is an escape character, you need another backslash to escape its own special functionality, so there it is, that&apos;s the final piece of the puzzle to produce valid checksums.
+</p>
+<p>
+  ## Special Thanks
+</p>
+<p>
+  Although I explained it all somewhat gracefully, that doesn&apos;t mean I managed to discover it myself sadly. I connected the dots later, but the actual information was provided to me by <em>Tim Green</em>, creator of [Steam Edit](https://steamedit.tg-software.com/). My application intends to be a cross-platform and open-source alternative to it, but in no way do I intend to make it obsolete or discredit <em>Tim</em>, he has been a great help in the creation of this program, and I&apos;m positive I wouldn&apos;t have been able to create it without his help. A big thanks to him.
+</p>
+<p>
+  I&apos;d also like to thank [<em>Leonid</em>](https://github.com/leovp) for indirectly aiding me by making the source code of [steamfiles](https://github.com/leovp/steamfiles) freely available to anyone, it has greatly helped me understand how the file worked, and I even borrowed some code.
+</p>
+<p>
+  Finally, thanks to [<em>xPaw</em>](https://github.com/xPaw) for documenting an update to the file format that included an extra checksum in [this](https://github.com/SteamDatabase/SteamAppInfo) repo.
+</p>
+</description>
+
+<launchable type="desktop-id">steam-metadata-editor.desktop</launchable>
```

Comment:
this is incorrect, it doesn't match the installed desktop file.

Reply:
What should I put there?

Reply:
what file name do you give when installing the .desktop file? That's what yyou should have here.

Reply:
Oh, but that depends on the FLATPAK_ID variable, I assume that resolves to `io.github.tralph3.Steam_Metadata_Editor` right?

Reply:
I verified that yes, it does.